docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -350,6 +350,282 @@ and a file system location of ``/var/my_tables/test_table``::
+    - The size of all the files under the partition
+  * - ``data``
+    - ``row(... row (min ..., max ... , null_count bigint))``
+    - Partition range metadata
+
+``$files`` table
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``$files`` table provides a detailed overview of the data files corresponding to the  Iceberg table.
+
+Use the following query::
+
+    SELECT * FROM "test_table$files"
+
+to retrieve the information about the data files of the Iceberg table ``test_table``.
+
+.. code-block:: text
+
+     content  | file_path                                                                                                                     | record_count    | file_format   | file_size_in_bytes   |  column_sizes        |  value_counts     |  null_value_counts | nan_value_counts  | lower_bounds                |  upper_bounds               |  key_metadata  | split_offsets  |  equality_ids
+    ----------+-------------------------------------------------------------------------------------------------------------------------------+-----------------+---------------+----------------------+----------------------+-------------------+--------------------+-------------------+-----------------------------+-----------------------------+----------------+----------------+---------------
+     0        | hdfs://hadoop-master:9000/user/hive/warehouse/test_table/data/c1=3/c2=2021-01-14/af9872b2-40f3-428f-9c87-186d2750d84e.parquet |  1              |  PARQUET      |  442                 | {1=40, 2=40, 3=44}   |  {1=1, 2=1, 3=1}  |  {1=0, 2=0, 3=0}   | <null>            |  {1=3, 2=2021-01-14, 3=1.3} |  {1=3, 2=2021-01-14, 3=1.3} |  <null>        | <null>         |   <null>
+
+
+
+The output of the query has the following columns:
+
+.. list-table:: Partitions columns
+  :widths: 20, 30, 50
+  :header-rows: 1
+
+  * - Name
+    - Type
+    - Description
+  * - ``content``
+    - ``integer``
+    - TODO
+  * - ``file_path``
+    - ``varchar``
+    - The data file location
+  * - ``file_format``
+    - ``varchar``
+    - The format of the data file
+  * - ``record_count``
+    - ``bigint``
+    - The number of entries contained in the data file
+  * - ``file_size_in_bytes``
+    - ``bigint``
+    - The data file size
+  * - ``column_sizes``
+    - ``map(integer, bigint)``
+    - The size for each column in the columnar file
+  * - ``value_counts``
+    - ``map(integer, bigint)``
+    - The count of entries per column in columnar file
+  * - ``null_value_counts``
+    - ``map(integer, bigint)``
+    - The number of ``NULL`` values per column in the columnar file
+  * - ``nan_value_counts``
+    - ``map(integer, bigint)``
+    - The number of non numerical values per column in the columnar file
+  * - ``lower_bounds``
+    - ``map(integer, bigint)``
+    - The lower bounds per column in the columnar file
+  * - ``upper_bounds``
+    - ``map(integer, bigint)``
+    - The upper bounds per column in the columnar file
+  * - ``key_metadata``
+    - ``varbinary``
+    - TODO
```

Comment:
You may find inspiration in https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/ContentFile.java

Reply:
Thank you very much.